library/bigip_pool2.py

```diff
@@ -0,0 +1,631 @@
+      port: 80
+
+- name: Delete pool
+  bigip_pool:
+      server: "lb.mydomain.com"
+      user: "admin"
+      password: "secret"
+      state: "absent"
+      name: "my-pool"
+      partition: "Common"
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.f5_utils import *
+from netaddr import IPAddress, AddrFormatError
+
+
+class Parameters(AnsibleF5Parameters):
+
+    api_map = {
+        'loadBalancingMode': 'lb_method', 'slowRampTime': 'slow_ramp_time',
+        'reselectTries': 'reselect_tries',
+        'serviceDownAction': 'service_down_action'
+               }
+
+    updatables = [
+        'monitor_type', 'quorum', 'monitors', 'service_down_action',
+        'description', 'lb_method', 'host', 'port', 'slow_ramp_time',
+        'reselect_tries'
+                    ]
+
+    returnables = [
+        'monitor_type', 'quorum', 'monitors', 'service_down_action',
+        'description', 'lb_method', 'host', 'port', 'monitor',
+        'slow_ramp_time', 'reselect_tries', 'name', 'member_name'
+    ]
+
+    api_attributes = [
+         'description', 'name', 'loadBalancingMode', 'monitor', 'slowRampTime',
+         'reselectTries', 'serviceDownAction'
+    ]
+
+    @property
+    def lb_method(self):
+        rawvalue = self._values['lb_method']
+        if rawvalue is None:
+            return None
+        value = rawvalue.replace('_', '-')
+        return value
+
+    @property
+    def monitors(self):
+        monitors = self._values['monitors']
+        monitor_type = self._values['monitor_type']
+        error1 = "The 'monitor_type' parameter cannot be empty when " \
+                 "'monitors' parameter is specified."
+        error2 = "The 'monitor' parameter cannot be empty when " \
+                 "'monitor_type' parameter is specified"
+        if monitors is not None and monitor_type is None:
+            raise F5ModuleError(error1)
+        if monitors is None and monitor_type is not None:
+            raise F5ModuleError(error2)
+        return monitors
+
+    @property
+    def quorum(self):
+        value = self._values['quorum']
+        error = "Quorum value must be specified with monitor_type 'm_of_n'."
+        if self._values['monitor_type'] == 'm_of_n' and value is None:
+            raise F5ModuleError(error)
+        return value
+
+    @property
+    def monitor(self):
+        monitors = self.monitors
+        monitor_type = self._values['monitor_type']
+        quorum = self.quorum
+
+        if monitors is None:
+            return None
+
+        if monitor_type == 'and_list':
+            and_list = list()
+            for m in monitors:
+                if monitors.index(m) == 0:
+                    and_list.append(m)
+                else:
+                    and_list.append('and')
+                    and_list.append(m)
+            result = ' '.join(and_list)
+        if monitor_type == 'm_of_n':
+            min_list = list()
+            prefix = 'min' + ' ' + str(quorum) + ' ' + 'of' + ' ' + '{'
```

Comment:
change this to be a string template using `.format`

prefix = 'min {0} of {'.format(str(quorum))